From 2030d382b49a1b957de829a67f74d9cc127c55ee Mon Sep 17 00:00:00 2001
From: Peter Jones <pjones@redhat.com>
Date: Wed, 17 Oct 2012 17:48:44 -0400
Subject: [PATCH 25/36] [pesign] Only shut down nss in pesign.c if we're not
 the daemon.

The daemon does its own init and shutdown.

Signed-off-by: Peter Jones <pjones@redhat.com>
---
 src/pesign.c | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/src/pesign.c b/src/pesign.c
index be6494e..6c10b6d 100644
--- a/src/pesign.c
+++ b/src/pesign.c
@@ -804,6 +804,14 @@ main(int argc, char *argv[])
 	}
 	pesign_context_free(ctxp);
 
-	NSS_Shutdown();
+	if (!daemon) {
+		SECStatus status = NSS_Shutdown();
+		if (status != SECSuccess) {
+			fprintf(stderr, "could not shut down NSS: %s",
+				PORT_ErrorToString(PORT_GetError()));
+			exit(1);
+		}
+	}
+
 	return (rc < 0);
 }
-- 
1.7.12.1

